### PR TITLE
Update db

### LINKED ISSUE #60: Do things to update_time from Account, Shop, Product, etc

### LINKED ISSUE #65: update coupon table
- [x] Automatically delete expired coupons.
- [x] Ensure uniqueness of the pair (`coupon_code`, `shop_uuid`).
- [x] additional requested change: make coupon global.

### LINKED ISSUE #66: update comment table

### PR DESCRIPTION
**Pull request by Vghxv**
### Fixes
- #60 
- #65 
- #66

#### GET
- None

### Misc Changes
- None

### CODE REVIEW

**Comment by Vghxv** [comment]
note : this PR will break coupon api

**Comment by Dragonfly911117** [comment]
> note : this PR will break coupon api

Why not update them as well?

File: app/router/coupon.py [hunk 1]
```diff
@@ -95,38 +89,44 @@ async def create_coupon(
...
+
+        "coupon_code": "30OFFSHOP",
+        "discount": 20
+        """
+    if account.role != 1:
         raise HTTPException(
-            status_code=status.HTTP_400_BAD_REQUEST,
-            detail=f"Coupon code {coupon_form['coupon_code']} under shop {shop_uuid} doesn't exist."
+            status_code=status.HTTP_403_FORBIDDEN,
+            detail=f"Only admin can update coupon for other shops."
```
Comment (PEKKA1117) [inline app/router/coupon.py]: No need to use f-string here

File: app/utils/auth.py [hunk 2]
```diff
@@ -83,11 +83,6 @@ def get_account(uuid: str) -> Account | None:
         account = result[0]
         if account["birthday"]:
             account["birthday"] = account["birthday"].strftime("%Y-%m-%d")
```
Comment (PEKKA1117) [inline app/utils/auth.py]: Birthday is also in `date` type as well. This line is meaningless.

File: sql-scripts/7Comment.sql [hunk 3]
```diff
@@ -1,22 +1,37 @@
...
+        primary key,
+    product_uuid varchar(64)                         not null,
+    account_uuid varchar(64)                         not null,
+    text         varchar(512)                        not null,
+    likes        int                                 not null,
+    dislikes     int                                 not null,
+    update_time  timestamp default CURRENT_TIMESTAMP null on update CURRENT_TIMESTAMP,
     constraint Comment_ibfk_1
-        foreign key (transaction_uuid) references Transaction (transaction_uuid),
+        foreign key (product_uuid) references Product (product_uuid),
```
Comment (PEKKA1117) [inline sql-scripts/7Comment.sql:12]: I thought we are commenting to orders?

File: sql-scripts/8Coupon.sql [hunk 4]
```diff
@@ -3,18 +3,20 @@ create table Coupon
     shop_uuid   varchar(64)                         not null,
```
Comment (PEKKA1117) [inline sql-scripts/8Coupon.sql]: No `shop_uuid`s

**Review by Dragonfly911117 — CHANGES_REQUESTED** [review]

File: app/router/coupon.py (hunk 1 above)
Comment (PEKKA1117) [reply app/router/coupon.py]:
```suggestion
detail="Only admin can update coupon for other shops."
``

File: sql-scripts/7Comment.sql (hunk 3 above)
Comment (Vghxv) [reply sql-scripts/7Comment.sql:12]: I changed it to product, so comment and transaction willl not functionally dependent on (transaction_uuid, account_uuid)

File: sql-scripts/8Coupon.sql (hunk 4 above)
Comment (PEKKA1117) [reply sql-scripts/8Coupon.sql]:
```suggestion
```

File: app/utils/auth.py (hunk 2 above)
Comment (Dragonfly911117) [reply app/utils/auth.py:85]:
```suggesion
        account = result[0]
```

File: sql-scripts/8Coupon.sql (hunk 4 above)
Comment (Vghxv) [reply sql-scripts/8Coupon.sql]: not reload datagrid before export@@

**Review by Dragonfly911117 — APPROVED** [review]
LGTM

**Pull request merged**